Jadaoun

JadAoun1 opened the pull request:
Added the DeleteAccountPage design. 

Added the ChangePasswordPage design.

Added routes in App.js for ChangePassword and DeleteAccount pages.

Created button navigation in Settings.js for ChangePassword and DeleteAccount pages.

Added the RegistrationPage design.

Added the SignInPage design.

Added icon images into the assets folder.

JadAoun1 committed:
Update SignIn.js

JadAoun1 committed:
Update Registration.js

JadAoun1 committed:
Update Modal.js

JadAoun1 committed:
Update DeleteAccountModal.js

JadAoun1 committed:
Update DeleteAccountModal.js

JadAoun1 committed:
Update App.js

Added route for new DeleteAccountPage

JadAoun1 committed:
Update Settings.js

Adjusted DeleteAccount button to navigate to DeleteAccountPage on click

JadAoun1 committed:
Create ChangePassword.js

Creates a page for the user to change their password from the settings page.

JadAoun1 committed:
Update Settings.js

Adds a ChangePassword button to the right of the DeleteAccount button and adds a navigation to the ChangePassword page

JadAoun1 committed:
Update App.js

Adds a route to the ChangePassword page

JadAoun1 committed:
Update and rename DeleteAccountModal.js to DeleteAccountPage.js

Adjusts the minimum height of the pageContainer to ensure that the page covers the whole viewport

JadAoun1 committed:
Update ChangePassword.js

JadAoun1 committed:
Update DeleteAccountPage.js

Adjusts spacing of elements to leave less empty space vertically

racecn commented on src/components/user/ChangePassword.js:
```diff
@@ -0,0 +1,135 @@
+});
+
+const backgroundStyles = () => ({
+  backgroundColor: 'rgba(91, 46, 72, 1)',
```
we have to be careful adding these infile color changes, maybe we can make this one of the tailwind config colors so it can be used globally?

rishithab48 replied:
I agree. What is that color? On my branch, I made custom colors for our website theme in tailwind.config.js.

racecn commented on src/components/user/ChangePassword.js:
```diff
@@ -0,0 +1,135 @@
+
+
+const labelStyles = () => ({
+  color: 'rgba(255, 255, 255, 1)',
```
this is just white no?

JadAoun1 replied:
yes it is, i'll adjust it to just say white

JadAoun1 committed:
Update ChangePassword.js

racecn commented on src/components/user/ChangePassword.js:
```diff
@@ -0,0 +1,135 @@
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  padding: '10px 45px',
```
All of these values should be relative with em or rem so it can be reactive,

JadAoun1 committed:
Update ChangePassword.js

Simplified color values and added em/rem alignment values to make the page reactive.

racecn commented:
Also this file is not included in the assets 

Can't resolve '../../assets/logo_with_circle_border-removebg.png' 

Can we add this to the commit please?

JadAoun1 committed:
Add files via upload

JadAoun1 committed:
Update ChangePassword.js

Added color code for Maroon from Rishitha's tailwind.config.js branch code.

racecn commented on src/assets/trash-icon.png:
We have FontAwesome Icons that can be imported, they will be infinitely resizable and recolorable on demand as they are SVG format
ex.
`import { faHome, faUser, faPlus, faSignOutAlt, faSignInAlt, faGear } from '@fortawesome/free-solid-svg-icons';`

usage: 
`<FontAwesomeIcon className="w-6 h-6 mt-1 text-[#e3bd96]" icon={faSignOutAlt} />`
                    
you can browse them here if you'd like: [Icons]( https://fontawesome.com/search)

rishithab48 commented on src/components/user/Registration.js:
```diff
@@ -1,226 +1,146 @@
-  const [showPassword, setShowPassword] = useState(false); // State variable to track password visibility
+const SignUpPage = (props) => {
+  // Styles
+  const styles = {
```
can you explain me how this works? I thought we are using inline tailwind.

rishithab48 commented on src/components/user/Registration.js:
```diff
@@ -1,226 +1,146 @@
-  const confirmPasswordRef = useRef();
-  const usernameRef = useRef();
+  // JSX Structure with styles applied
+return (
```
I don't think you are importing the right picture for logo

JadAoun1 committed:
Update Registration.js

Handles account creation and redirects to user onboarding page

JadAoun1 committed:
Update tailwind.config.js

JadAoun1 committed:
Update App.js

JadAoun1 committed:
Update and rename src/components/shared/Modal.js to src/components/shHandleSignInClick.js

JadAoun1 committed:
Rename shHandleSignInClick.js to HandleSignInClick.js

JadAoun1 committed:
Update navbar.js

JadAoun1 committed:
Update ChangePassword.js

JadAoun1 committed:
Update DeleteAccountPage.js

JadAoun1 committed:
Update Registration.js

JadAoun1 committed:
Update SignIn.js

JadAoun1 committed:
Update Home.js

JadAoun1 committed:
Create HandleSignInClick.js

JadAoun1 committed:
Update HandleSignInClick.js

racecn commented on src/pages/Home.js:
Make sure you pull from develop often, this home version is far behind the current version

racecn commented on src/components/user/ChangePassword.js:
try to use tailwind or create a button style in the tailwind config so we can edit it easily later

racecn commented on src/components/user/DeleteAccountPage.js:
```diff
@@ -0,0 +1,168 @@
+});
+
+// Trash Icon Style
+const trashIconStyle = () => ({
```
The filter to set the color of the icon would be unnecessary if we used FontAwesome Icons instead

racecn commented on src/components/user/DeleteAccountPage.js:
```diff
@@ -0,0 +1,168 @@
+  // Navigate back to home after deletion
+  const handleDelete = () => {
+      console.log("Account is being deleted...");
+      // Add deletion logic here
```
we have to ensure that logic is implemented fully before a pull request to the main branch is done

racecn commented on src/components/user/SignIn.js:
```diff
@@ -1,65 +1,196 @@
+
+  // Adjusted sign in box styling for vertical coverage
+  const signInBoxStyle = {
+    maxWidth: '600px',
+    width: '100%',
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'space-around', // Ensures even distribution of space
+    alignItems: 'center',
+    position: 'relative',
+    overflow: 'hidden',
+    paddingTop: '20px', // Padding at the top
+    paddingBottom: '20px', // Padding at the bottom
```
Unsure why we are using pixel specific styling again instead of reactive content that would allow for window resizing and mobile app possibility in the future

racecn commented on src/components/user/SignIn.js:
```diff
@@ -1,65 +1,196 @@
+
+  // Separate style for the sign-up button
+  const signUpButtonStyle = {
+    backgroundColor: '#ffffff',
+    width: '300px',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    padding: '10px',
+    borderRadius: '40px',
+    marginBottom: '40px',
+    marginTop: '1px',
+    boxShadow: '0px 4px 8px rgba(0, 0, 0, 0.8)',
```
Same issue that was present in the last review of pull request on ffffff is white and pixel-specific stuff

racecn commented on src/components/user/SignIn.js:
```diff
@@ -1,65 +1,196 @@
-        <button type="button" disabled={loading} onClick={handleSignInWithEmailAndPassword}>
-          Sign In
+        {/* Divider text */}
+        <span style={textStyle(20, 400, '20px')}>---------- OR ----------</span>
```
can we use a styled line instead of the ascii art?

JadAoun1 committed:
Update ChangePassword.js

Changed to in-line styling and utilizes font-awesome icon rather than an imported svg image.

Also utilizes rem spacing in order to make the page more responsive.

JadAoun1 committed:
Update DeleteAccountPage.js

Uses in-line styling and replaces .svg icon with font-awesome icon.

Also uses rem spacing units instead of pixel spacing units in order to make the page more responsive.

JadAoun1 committed:
Update Registration.js

Changes styling to in-line.

Utilizes rem units instead of px.

Adds a check to make sure all fields are filled before allowing user to click "Create Account."

No longer uses color codes.

JadAoun1 committed:
Update Registration.js

Fixes bug with handleSignUpWithPopup.

JadAoun1 committed:
Update Registration.js

Adjusts positioning of eye slash icons.

JadAoun1 committed:
Update Registration.js

Creates account and handles firebase storage.

JadAoun1 committed:
Create ArtistProfilePage.js

JadAoun1 committed:
Create ReviewerProfilePage.js

rishithab48 commented on src/components/user/ChangePassword.js:
I just checked the page. The styling is off. Can you fix it?

JadAoun1 committed:
Merge branch 'develop' of https://github.com/WSU-4110/Novelology into jadaoun

JadAoun1 committed:
Update ChangePassword.js

JadAoun1 committed:
Update DeleteAccountPage.js

JadAoun1 committed:
Update App.js

JadAoun1 committed:
Update Settings.js

JadAoun1 committed:
Merge branch 'develop' of https://github.com/WSU-4110/Novelology into jadaoun

JadAoun1 committed:
None.

JadAoun1 committed:
Reverting changes.

JadAoun1 committed:
Reverting changes.
